Demo/src/main/java/com/braintreepayments/demo/ShopperInsightsFragment.kt

```diff
@@ -36,48 +50,123 @@ class ShopperInsightsFragment : BaseFragment() {
         actionButton = view.findViewById(R.id.actionButton)
+        payPalVaultButton = view.findViewById(R.id.payPalVaultButton)
+        venmoButton = view.findViewById(R.id.venmoButton)
         emailInput = view.findViewById(R.id.emailInput)
         countryCodeInput = view.findViewById(R.id.countryCodeInput)
         nationalNumberInput = view.findViewById(R.id.nationalNumberInput)
         emailNullSwitch = view.findViewById(R.id.emailNullSwitch)
         phoneNullSwitch = view.findViewById(R.id.phoneNullSwitch)
+
+        emailInput.editText?.setText("PR1_merchantname@personal.example.com")
+        nationalNumberInput.editText?.setText("4082321001")
+        countryCodeInput.editText?.setText("1")
     }
 
-    private fun setupActionButton() {
-        actionButton.setOnClickListener {
-            val email =
-                if (emailNullSwitch.isChecked) null else emailInput.editText?.text.toString()
-            val countryCode =
-                if (phoneNullSwitch.isChecked) null else countryCodeInput.editText?.text.toString()
-            val nationalNumber =
-                if (phoneNullSwitch.isChecked) null else nationalNumberInput.editText?.text.toString()
-
-            val request = if (countryCode != null && nationalNumber != null) {
-                ShopperInsightsRequest(email, ShopperInsightsBuyerPhone(countryCode, nationalNumber))
-            } else {
-                ShopperInsightsRequest(email, null)
-            }
-            shopperInsightsClient.getRecommendedPaymentMethods(request) { result ->
-                responseTextView.text = when (result) {
-                    is ShopperInsightsResult.Success -> {
-                        "PayPal Recommended ${result.response.isPayPalRecommended} " +
-                                "\n Venmo Recommended ${result.response.isVenmoRecommended}"
-                    }
-                    is ShopperInsightsResult.Failure -> result.error.toString()
+    private fun fetchShopperInsights() {
+        val email =
+            if (emailNullSwitch.isChecked) null else emailInput.editText?.text.toString()
+        val countryCode =
+            if (phoneNullSwitch.isChecked) null else countryCodeInput.editText?.text.toString()
+        val nationalNumber =
+            if (phoneNullSwitch.isChecked) null else nationalNumberInput.editText?.text.toString()
+
+        val request = if (countryCode != null && nationalNumber != null) {
+            ShopperInsightsRequest(email, ShopperInsightsBuyerPhone(countryCode, nationalNumber))
+        } else {
+            ShopperInsightsRequest(email, null)
+        }
+
+        shopperInsightsClient.getRecommendedPaymentMethods(
+            request
+        ) { result ->
+            when (result) {
+                is ShopperInsightsResult.Success -> {
+                    payPalVaultButton.isEnabled = result.response.isPayPalRecommended
+                    venmoButton.isEnabled = result.response.isVenmoRecommended
```

Comment:
Would we want to show/hide the button based on if the payment method is recommended instead of setting it to enabled/disabled? I was thinking that merchants would probably show/hide the button for their integration.

Reply:
Definitely - a merchant should not be disabling the Venmo or PayPal buttons 😆. Maybe we can 🏈 punt that for another PR and align with how iOS shows the buttons?

Reply:
Sounds good!